### PR TITLE
Silence output when optimistically checking for fast-register archive

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
As an optimization, fast-register checks to see if a remote distribution exists before uploading it. Confusingly this prints a 404 in the reasonable case where the distribution does not exist. Annoyingly, if aws retries in a flytekit config are set to > 1 this 404 prints multiple times and is pretty alarming. This change silences the output of that command.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
https://docs.google.com/document/d/1n7DmKXaOv8dAb9eHD4TKqka6FsdJNs3Tugpf1I6nvMg/edit?usp=sharing

## Tracking Issue
https://github.com/lyft/flyte/issues/554

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — COMMENTED** [review]
is this the right approach?  redirecting to dev/null seems kinda heavy handed.  Also, is there a way to not retry four times?  are you referring to the awscli setting?
